sdk/python/kfp/dsl/types/type_annotations.py

```diff
@@ -220,15 +220,26 @@ def get_short_type_name(type_name: str) -> str:
       typing.Dict[str, str] -> Dict
       List -> List
       str -> str
+      system.Artifact@0.0.1 -> Artifact
+      system.Dataset@0.0.1 -> Dataset
+      google.VertexModel@0.0.1 -> VertexModel
 
     Args:
       type_name: The original type name.
 
     Returns:
       The short form type name or the original name if pattern doesn't match.
     """
-    match = re.match('(typing\.)?(?P<type>\w+)(?:\[.+\])?', type_name)
-    return match['type'] if match else type_name
+    parameter_match = re.match('(typing\.)?(?P<type>\w+)(?:\[.+\])?', type_name)
+    artifact_match = re.match('(system\.|google\.)(?P<type>\w+)@(\d\.\d\.\d)',
```

Comment:
I think there might be a simpler way to extract the artifact type: https://github.com/kubeflow/pipelines/blob/7bd31d104bd403a830bf2a455c9c2c0dbf493c4d/sdk/python/kfp/local/importer_handler.py#L137-L142

I also wonder if we need to at all in some places in this PR. Can we just pass the type from the `PipelineArtifactChannel` directly to the `LoopArtifactArgument`?

Reply:
This method doesnt handle the schema version `@0.0.1` so we would have to pre-parse the string every time to check the channel type and it wouldn't support any gcpc artifact types.

Also the usage is a little different bc that method presupposes the input is a valid artifact class and always returns an artifact class, whereas `type_annotations.get_short_type_name` is more general and works for all input types.

My preference here is to keep `type_annotations.get_short_type_name`.

Reply:
Is this update still needed with the changed implementation? If I remove this change, tests still pass.

Reply:
Removed.